src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -55,6 +105,35 @@ pub fn prebuilt_gcc_config(builder: &Builder<'_>, target: TargetSelection) -> Gc
         )
     });
 
+    // Try to download GCC from CI if configured and available
+    if let GccCiMode::DownloadFromCi = &builder.config.gcc_ci_mode {
+        if builder.build.build == "x86_64-unknown-linux-gnu" {
+            let sha = detect_gcc_sha(
+                &builder.config,
+                builder.config.rust_info.is_managed_git_subrepository(),
+            );
+            let root = ci_gcc_root(&builder.config);
+            let gcc_stamp = BuildStamp::new(&root).with_prefix("gcc").add_stamp(&sha);
+            if !gcc_stamp.is_up_to_date() && !builder.config.dry_run() {
+                builder.config.download_ci_gcc(&sha, &root);
+                t!(gcc_stamp.write());
+            }
+            // FIXME: put libgccjit.so into a lib directory in dist::Gcc
+            return GccBuildStatus::AlreadyBuilt(root.join("libgccjit.so"));
+        } else {
+            eprintln!(
+                "GCC CI download is only available for the `x86_64-unknown-linux-gnu` target"
+            );
+        }
```

Comment:
This looks more readable I think

```suggestion
        if builder.build.build != "x86_64-unknown-linux-gnu" {
            eprintln!(
                "GCC CI download is only available for the `x86_64-unknown-linux-gnu` target"
            );
        }

        let sha = detect_gcc_sha(
            &builder.config,
            builder.config.rust_info.is_managed_git_subrepository(),
        );
        let root = ci_gcc_root(&builder.config);
        let gcc_stamp = BuildStamp::new(&root).with_prefix("gcc").add_stamp(&sha);
        if !gcc_stamp.is_up_to_date() && !builder.config.dry_run() {
            builder.config.download_ci_gcc(&sha, &root);
            t!(gcc_stamp.write());
        }
        // FIXME: put libgccjit.so into a lib directory in dist::Gcc
        return GccBuildStatus::AlreadyBuilt(root.join("libgccjit.so"));
```